Support rowClassRules/cellClassRules configs

amcclain opened the pull request:
+ Add as top-level configs to GridModel and Column, respectively.
+ Throw if detected within agOptions to ensure migration.
+ Remove obsolete `GridModel.compact` config - deprecated in v29.
+ Fixes #1331 

Hoist P/R Checklist
-------------------

**Pull request authors:** Review and check off the below. Items that do not apply can also be
checked off to indicate they have been considered. If unclear if a step is relevant, please leave
unchecked and note in comments.

- [x] Caught up with `develop` branch as of last change.
- [x] Added CHANGELOG entry, or determined not required.
- [x] Reviewed for breaking changes, added `breaking-change` label + CHANGELOG if so.
- [x] Updated doc comments / prop-types, or determined not required.
- [x] Reviewed and tested on Mobile, or determined not required.
- [x] Created Toolbox branch / PR, or determined not required.

**If your change is still a WIP**, please use the "Create draft pull request" option in the split
button below to indicate it is not ready yet for a final review.

> **Pull request reviewers:** when merging this P/R, please consider using a *squash commit* to
> collapse multiple intermediate commits into a single commit representing the overall feature
> change. This helps keep the commit log clean and easy to scan across releases. PRs containing a
> single commit should be *rebased* when possible.

amcclain commented:
See also https://github.com/xh/toolbox/pull/509 for TB demo of `cellClassRules` - need to find a reasonable place to use `rowClassRules` also (didn't come up with anything I really liked within SampleGrid, which is getting very busy - can look elsewhere).

Main decision points I wanted to discuss before finalizing:
- Are we OK just going with the agParams and not trying to normalize the signatures of the rule functions? (I.e. by transforming to pass in record, store, gridModel, etc as we typically do)?
- OK to throw if either config provided via agOptions?
- OK to maintain support for pre-existing `rowClassFn` and `cellClass` configs in addition to new configs?

cnrudd reviewed (approved):
Does this need a changelog entry?

Otherwise all good.

amcclain commented:
I added a changelog entry to move this forward in its current configuration.  Always happy to discuss, but I think this strikes the best balance:

1. Keeps the shape of the objects the same as it was for direct agGrid usage - should make it easy for apps to migrate
2. Keeps the apiDeprecated check in place, which will throw on existing app usages within agOptions, but calls out this caveat in the changelog. Given (1) I think this is reasonable to keep as a "minor breaking change" without triggering a new major version - use of `agOptions` by apps always carries caveats.
3. We could consider removing the alternate configs in a future release, but I don't think we need to force the issue now, and ag-Grid supports both.  The pre-existing ones are easier to use if you aren't worried about data changing and requiring them to be applied and removed within a session.
